### ISSUE TITLE
SIGSEGV encountered - SM80 with CUDA11

### ISSUE THREAD

**Issue by RadoZvolensky** [no-fee-ethereum-mining/nsfminer#83]
**Describe the bug**
Once in a while the miner will crash with folowing segmentation faults:

```
SIGSEGV encountered ...
stack trace:
backtrace() returned 18 addresses
[0x55f7ec22b992]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f74bb58c210]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x1567f0) [0x7f74b994e7f0]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21bc3e) [0x7f74b9a13c3e]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x364e4f) [0x7f74b9b5ce4f]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21c3a5) [0x7f74b9a143a5]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11c872) [0x7f74b9914872]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11dd76) [0x7f74b9915d76]
/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x73) [0x7f74b9ae1943]
[0x55f7ec53bfc9]
[0x55f7ec517ff5]
[0x55f7ec5541a9]
[0x55f7ec502cfa]
[0x55f7ec5024f6]
[0x55f7ec2bd5d2]
[0x55f7ec605a84]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x9609) [0x7f74bb8ad609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f74bb668293]
```

**worker thread**
```
SIGSEGV encountered ...
stack trace:
backtrace() returned 18 addresses
 [97mcu-0  [0mUsing 4.14 GB out of 39.59 GB GPU memory
 [97mcu-1  [0mUsing 4.14 GB out of 39.59 GB GPU memory
[0x5557a1fec992]
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f69b65dc210]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x1567f0) [0x7f69b499e7f0]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21bc3e) [0x7f69b4a63c3e]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x364e4f) [0x7f69b4bace4f]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21c3a5) [0x7f69b4a643a5]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11c872) [0x7f69b4964872]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11dd76) [0x7f69b4965d76]
/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x73) [0x7f69b4b31943]
[0x5557a22fcfc9]
[0x5557a22d8ff5]
[0x5557a23151a9]
[0x5557a22c3cfa]  [91mcu-3  [0mException thrown in Worker thread: GPU error: CUDA error in func search at line 349 driver shutting down
```
**initEpoch**
```
SIGSEGV encountered ...
  [97mcu-3  [0mUsing 4.14 GB out of 39.59 GB GPU memory
stack trace:
  [97mcu-3  [0mUnexpected error CUDA error in func initEpoch at line 117 driver shutting down on CUDA device 0000:c1:00.0
  [97mcu-3  [0mMining suspended ...
  [91mcu-3  [0mException thrown in Worker thread: GPU error: CUDA error in func workLoop at line 194 driver shutting down
  [91mcu-3  [0mTerminating due to --exit
backtrace() returned 18 addresses
```
```
SIGSEGV encountered ...
stack trace:
  [97mcu-0  [0mUsing 4.14 GB out of 39.59 GB GPU memory
  [97mcu-3  [0mUsing 4.14 GB out of 39.59 GB GPU memory
  [97mcu-2  [0mUsing 4.14 GB out of 39.59 GB GPU memory
  [97mcu-3  [0mUnexpected error CUDA error in func initEpoch at line 117 driver shutting down on CUDA device 0000:c1:00.0
  [97mcu-3  [0mMining suspended ...
  [97mcu-2  [0mUnexpected error CUDA error in func initEpoch at line 117 driver shutting down on CUDA device 0000:81:00.0
  [97mcu-2  [0mMining suspended ...
backtrace() returned 18 addresses
  [91mcu-3  [0mException thrown in Worker thread: GPU error: CUDA error in func workLoop at line 194 driver shutting down
  [91mcu-2  [0mException thrown in Worker thread: GPU error: CUDA error in func workLoop at line 194 driver shutting down
  [91mcu-2  [0mTerminating due to --exit
  [91mcu-3  [0mTerminating due to --exit
```
**func search**
```
SIGSEGV encountered ...
stack trace:
  [97mcu-2  [0mUsing 4.14 GB out of 39.59 GB GPU memory
backtrace() returned 18 addresses
  [91mcu-1  [0mException thrown in Worker thread: GPU error: CUDA error in func search at line 349 driver shutting down
  [91mcu-1  [0mTerminating due to --exit
  [91mcu-3  [0mException thrown in Worker thread: GPU error: CUDA error in func search at line 349 driver shutting down
  [91mcu-3  [0mTerminating due to --exit
[0x55de93d69992]

  [91mmain  [0mGot interrupt ...
/lib/x86_64-linux-gnu/libc.so.6(+0x46210) [0x7f6cef02c210]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x1567f0) [0x7f6ced3ee7f0]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21bc3e) [0x7f6ced4b3c3e]
  [91mmain  [0mGot interrupt ...

/lib/x86_64-linux-gnu/libcuda.so.1(+0x364e4f) [0x7f6ced5fce4f]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x21c3a5) [0x7f6ced4b43a5]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11c872) [0x7f6ced3b4872]
/lib/x86_64-linux-gnu/libcuda.so.1(+0x11dd76) [0x7f6ced3b5d76]
/lib/x86_64-linux-gnu/libcuda.so.1(cuMemcpyHtoD_v2+0x73) [0x7f6ced581943]
[0x55de94079fc9]
[0x55de94055ff5]
[0x55de940921a9]
[0x55de94040cfa]
[0x55de940404f6]
[0x55de93dfb5d2]
[0x55de94143a84]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x9609) [0x7f6cef34d609]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x43) [0x7f6cef108293]
```

**To Reproduce**
This bug can't be reproduced manually.

**Expected behavior**
There should not be segmentation faults and the miner should not access restricted area of memory, or memory that was already released.

**Environment (please complete the following information):**
- Ubuntu 20.04.2 LTS (Focal Fossa) Kernel: 5.4.0-65-generic
- 4 x NVIDIA A100 40GB
- Ethminer Version 1.3.3
- Ethminer options used
--exit --cuda --HWMON 1 --tstop 69 --tstart 65 --retry-delay 5 -P stratum2+tcp://<wallet_address>.<rig_name>@daggerhashimoto.eu.nicehash.com:3353 >> log.log 2>&1

**Additional context**
This should not be caused by overheating, the cards are underclocked & power limited. They are basically bored, holding 60 °C :)

**Comment by lesjokolat** [comment]
Could it be swap space is filling up.  I would make your swap at least as big as your ram  to see if that helps.

**Comment by jean-m-cyr** [comment]
Since you are mining at a stratum2 pool, it help to know if these crashes occur around epoch or difficulty changes by the pool.

**Comment by jean-m-cyr** [comment]
This should be corrected by PR #86. Reopen if not so.
